packages/edit-navigation/src/components/toolbar/index.js

```diff
@@ -1,19 +1,54 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
 /**
  * WordPress dependencies
  */
 import { Spinner } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { NavigableToolbar, BlockToolbar } from '@wordpress/block-editor';
+import { useEffect, useRef } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+function stopPropagation( event ) {
+	event.stopPropagation();
+}
+
+// Hack: Stop useBlockSelectionClearer from clearing block selection when the
+// block toolbar is clicked.
```

Comment:
Hm, how's the block editor dealing with this?

Reply:
I think because the top toolbar is rendered outside of the editor canvas it won't receive clicks. Similarly the non-top toolbar uses a slot to also render outside of the canvas, so both don't suffer from this issue.

We can probably fix in the same way, and I think we were going to look at [removing the fixed toolbar](https://github.com/WordPress/gutenberg/pull/28689#issuecomment-773018163), which should do that.